lib/rouge/lexers/json.rb

```diff
@@ -12,19 +12,55 @@ class JSON < RegexLexer
                 'application/hal+json'
 
       state :root do
-        rule /\s+/m, Text::Whitespace
-        rule /"/, Str::Double, :string
-        rule /(?:true|false|null)\b/, Keyword::Constant
-        rule /[{},:\[\]]/, Punctuation
-        rule /-?(?:0|[1-9]\d*)\.\d+(?:e[+-]?\d+)?/i, Num::Float
-        rule /-?(?:0|[1-9]\d*)(?:e[+-]?\d+)?/i, Num::Integer
+        rule /\s+/, Text::Whitespace
+        rule /{/, Punctuation, :object
+        rule /\[/, Punctuation, :array
+      end
+
+      state :object do
+        rule /\s+/, Text::Whitespace
+        rule /"/, Str::Double, :key
+        rule /:/, Punctuation, :value
+        rule /,/, Punctuation
+        rule /}/, Punctuation, :pop!
       end
 
-      state :string do
+      state :value do
+        rule /"/, Str::Char, :stringvalue
+        mixin :constants
+        rule /\s*?(?=})/, Text::Whitespace, :pop!
+        rule /\[/, Punctuation, :array
+        rule /{/, Punctuation, :object
+        rule /}/, Punctuation, :pop!
+        rule /,/, Punctuation, :pop!
+        rule /\s+/, Text::Whitespace
+      end
+
+      state :key do
```

Comment:
Based on [the description](https://www.json.org) on the JSON homepage, I think the more accurate label for this state would `:name`.